charmcraft/providers.py

```diff
@@ -112,26 +106,29 @@ def ensure_provider_is_available() -> None:
 
     :raises CommandError: if provider is not available.
     """
-    if is_provider_available():
-        return
-
-    if confirm_with_user(
-        "LXD is required, but not installed. Do you wish to install LXD "
-        "and configure it with the defaults?",
-        default=False,
-    ):
-        try:
-            lxd_installer.install()
-        except LXDInstallationError as error:
+    if not is_provider_available():
+        if confirm_with_user(
+            "LXD is required, but not installed. Do you wish to install LXD "
+            "and configure it with the defaults?",
```

Comment:
Not worth for this branch, but for the future: for readability in these cases please write the long message outside the `if`. Thanks!